BlazarData/src/main/java/com/hubspot/blazar/data/dao/RepositoryBuildDao.java

```diff
@@ -44,6 +48,10 @@
       "WHERE b.id = :branchId")
   BuildNumbers getBuildNumbers(@Bind("branchId") int branchId);
 
+  // Adding `ORDER BY buildNumber` here causes a filesort to occur
+  @SqlQuery("SELECT * FROM repo_builds WHERE branchId = :branchId AND state in (<state>)")
+  Set<RepositoryBuild> getRepositoryBuildsByState(@Bind("branchId") int branchId, @BindIn("states") List<RepositoryBuild.State> states);
```

Comment:
@gchomatas this is the query we're using to get the `queuedBuilds`.
The explain for this query shows this `using where`, and runs in ~24ms from my laptop.